src/cata_tiles.cpp

```diff
@@ -1680,7 +1680,7 @@ bool cata_tiles::draw_from_id_string(std::string id, TILE_CATEGORY category,
         // use a fair mix function to turn the "random" seed into a random int
         // taken from public domain code at http://burtleburtle.net/bob/c/lookup3.c 2015/12/11
 #define rot32(x,k) (((x)<<(k)) | ((x)>>(32-(k))))
-        unsigned int a = seed, b = -seed, c = seed*seed;
+        unsigned int a = seed, b = 0xFFFFFFFFUL-seed+1, c = seed*seed;
```

Comment:
Some suggestions here:
1. Consider uint32_t to make it clear that we're doing bitwise stuff.
2. Use UINT32_MAX instead of the magic number.
3. Spaces around binary operators as per coding style.